File: 00_legacy_materials/arifOS-upstream/packages/arifos-types/src/verdict.ts
/**
 * Verdict — Constitutional verdict types
 * DITEMPA BUKAN DIBERI
 */

export enum Verdict {
  PROCEED = "PROCEED",   // Low risk, constitution satisfied
  HOLD = "HOLD",         // Elevated risk, human review required (888_HOLD)
  BLOCK = "BLOCK",       // Critical risk, constitution violated
  SEAL = "SEAL",         // Terminal verdict, VAULT999 recorded
  VOID = "VOID",         // Action not permitted, constitution breach
}

export enum VerdictStatus {
  PENDING = "PENDING",
  APPROVED = "APPROVED",
  REJECTED = "REJECTED",
  EXPIRED = "EXPIRED",
}

export function isTerminalVerdict(v: Verdict): boolean {
  return [Verdict.SEAL, Verdict.BLOCK, Verdict.VOID].includes(v);
}

export function requiresHumanReview(v: Verdict): boolean {
  return v === Verdict.HOLD;
}

File: packages/arifos-types/py/arifos_types/verdict.py
"""
Verdict — Constitutional verdict types
DITEMPA BUKAN DIBERI
"""

from enum import Enum


class Verdict(str, Enum):
    PROCEED = "PROCEED"
    HOLD = "HOLD"
    BLOCK = "BLOCK"
    SEAL = "SEAL"
    VOID = "VOID"


class VerdictStatus(str, Enum):
    PENDING = "PENDING"
    APPROVED = "APPROVED"
    REJECTED = "REJECTED"
    EXPIRED = "EXPIRED"


TERMINAL_VERDICTS = {Verdict.SEAL, Verdict.BLOCK, Verdict.VOID}


def isTerminalVerdict(v: Verdict) -> bool:
    return v in TERMINAL_VERDICTS


def requiresHumanReview(v: Verdict) -> bool:
    return v == Verdict.HOLD